kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -510,25 +527,33 @@ def kube_job_failed(self, metric, scraper_config):
 
     def kube_job_status_failed(self, metric, scraper_config):
         for sample in metric.samples:
+            job_ts = 0
             tags = [] + scraper_config['custom_tags']
             for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
                 if label_name == 'job' or label_name == 'job_name':
                     trimmed_job = self._trim_job_tag(label_value)
+                    job_ts = self._extract_job_timestamp(label_value)
                     tags.append(self._format_tag(label_name, trimmed_job, scraper_config))
                 else:
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
-            self.job_failed_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
+            if job_ts != 0 and job_ts > self.failed_job_counts[frozenset(tags)].last_job_ts:
```

Comment:
We're iterating over `metric.samples` here. Are they always ordered by job timestamp? Otherwise we might have an issue here: if more than one job happened since last crawl, and we see the one with the most recent timestamp first, we're going to skip the older (but still new) one.

Reply:
btw this is the tricky part of this feature, so i would feel better if you added a unit test to validate it.